### PR TITLE
Minor suggestions and replaced use of obsolete Date-API

### PR DESCRIPTION
**Pull request by Peldan**

### CODE REVIEW

**Comment by Peldan** [comment]
> Please, read and sign the above mentioned agreement if you want to contribute to this project

Sorry, I have signed the agreement now.

File: ecs-logging-core/src/main/java/co/elastic/logging/TimestampSerializer.java [hunk 1]
```diff
@@ -110,9 +112,10 @@ private void serializeWithLeadingZero(StringBuilder builder, long value, int min
         private final long endOfCachedDate;
 
         private CachedDate(long epochTimestamp) {
-            SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.ROOT);
-            dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
-            cachedDateIso = dateFormat.format(new Date(epochTimestamp));
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+                    .localizedBy(Locale.ROOT)
+                    .withZone(ZoneId.of("UTC"));
+            cachedDateIso = formatter.format(Instant.ofEpochMilli(epochTimestamp));
```
Comment (felixbarny) [inline ecs-logging-core/src/main/java/co/elastic/logging/TimestampSerializer.java:118]: Uses APIs that are not available in Java 6

File: log4j2-ecs-layout/src/main/java/co/elastic/logging/log4j2/EcsLayout.java [hunk 2]
```diff
@@ -51,15 +51,16 @@
...
 import java.util.List;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 
 @Plugin(name = "EcsLayout", category = Node.CATEGORY, elementType = Layout.ELEMENT_TYPE)
 public class EcsLayout extends AbstractStringLayout {
 
-    public static final Charset UTF_8 = Charset.forName("UTF-8");
-    public static final String[] JSON_FORMAT = {"JSON"};
+    public static final Charset UTF_8 = StandardCharsets.UTF_8;
```
Comment (felixbarny) [inline log4j2-ecs-layout/src/main/java/co/elastic/logging/log4j2/EcsLayout.java:62]: Uses APIs that are not available in Java 6

File: log4j2-ecs-layout/src/test/java/co/elastic/logging/log4j2/ParameterizedStructuredMessage.java [hunk 3]
```diff
@@ -31,14 +31,12 @@
 public class ParameterizedStructuredMessage extends MapMessage<ParameterizedStructuredMessage, Object> {
 
     private static final String MESSAGE = "message";
-    private static final TriConsumer<String, Object, StringBuilder> ADD_KEY_VALUE_PAIR = new TriConsumer<>() {
-        @Override
-        public void accept(String key, Object value, StringBuilder builder) {
-            if (!MESSAGE.equals(key)) {
-                builder.append(' ').append(key).append('[').append(value).append(']');
-            }
+    private static final TriConsumer<String, Object, StringBuilder> ADD_KEY_VALUE_PAIR = (key, value, builder) -> {
```
Comment (felixbarny) [inline log4j2-ecs-layout/src/test/java/co/elastic/logging/log4j2/ParameterizedStructuredMessage.java:34]: Lambdas not available in Java 6

File: logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java [hunk 4]
```diff
@@ -34,20 +34,21 @@
...
 import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
 
 public class EcsEncoder extends EncoderBase<ILoggingEvent> {
 
-    private static final Charset UTF_8 = Charset.forName("UTF-8");
+    private static final Charset UTF_8 = StandardCharsets.UTF_8;
```
Comment (felixbarny) [inline logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java:44]: Uses APIs that are not available in Java 6

File: ecs-logging-core/src/main/java/co/elastic/logging/TimestampSerializer.java (hunk 1 above)
Comment (Peldan) [reply ecs-logging-core/src/main/java/co/elastic/logging/TimestampSerializer.java]: Hi, sorry about that. I've reverted the incompatible changes.

/Peldan

File: logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java [hunk 5]
```diff
@@ -47,7 +47,7 @@
     private boolean includeMarkers = false;
     private ThrowableProxyConverter throwableProxyConverter;
     private boolean includeOrigin;
-    private List<Pair> additionalFields = new ArrayList<Pair>();
+    private final List<Pair> additionalFields = new ArrayList<>();
```
Comment (felixbarny) [inline logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java:50]: This fails the build as diamond operators are not allowed when the bytecode level is 6.
```suggestion
    private final List<Pair> additionalFields = new ArrayList<Pair>();
```

File: log4j-ecs-layout/src/test/java/co/elastic/logging/log4j/ListAppender.java [hunk 6]
```diff
@@ -31,7 +31,7 @@
 import java.util.List;
 
 class ListAppender extends AppenderSkeleton {
-    private List<String> logEvents = new ArrayList<>();
+    private final List<String> logEvents = new ArrayList<>();
```
Comment (felixbarny) [inline log4j-ecs-layout/src/test/java/co/elastic/logging/log4j/ListAppender.java:34]:
```suggestion
    private final List<String> logEvents = new ArrayList<String>();
```

**Pull request merged**